src/features/courses/course-controller.ts

```diff
@@ -3143,12 +3152,17 @@ class CourseController {
     
                 // const versionAverage = (incoming.instance.averageScore * incoming.instance.numAttempts + incoming.questionResponse.problem_result.score)/(incoming.instance.numAttempts + 1);
     
+                if (studentTopicAssessmentInfo.numAttempts === 0) {
+                    // this is the student's first submission for this version -- 
+                    // update StudentGrade numAttempts to reflect the number of versions that were actually attempted
+                    result.grade.numAttempts++;
```

Comment:
shouldn't this just be result.grade.numAttempts = studentTopicAssessmentInfo.numAttempts (after it's been incremented)?

Reply:
Why does it only increment on the first submission?

Reply:
StudentGrade numAttempts to reflect the number of versions that were actually attempted